bigchaindb/backend/rethinkdb/config.py

```diff
@@ -0,0 +1,48 @@
+"""Database configuration functions."""
+import logging
+from functools import partial
+
+import rethinkdb as r
+
+from bigchaindb.backend import config
+from bigchaindb.backend.exceptions import DatabaseOpFailedError
+from bigchaindb.backend.rethinkdb.connection import RethinkDBConnection
+
+logger = logging.getLogger(__name__)
+
+TABLES = ('bigchain', 'backlog', 'votes')
+
+
+def _configure_table_helper(connection, table, *, replicas=None, shards=None):
+    try:
+        connection.run(
+            r.table(table).reconfigure(shards=shards, replicas=replicas)
+        )
+    except r.ReqlOpFailedError as e:
+        raise DatabaseOpFailedError from e
+
+
+def _configure_table(connection, table, *, replicas=None, shards=None):
+    if replicas is None:
+        table_config = connection.run(r.table(table).config())
+        replicas = len(table_config['shards'][0]['replicas'])
+    if shards is None:
+        table_config = connection.run(r.table(table).config())
+        shards = len(table_config['shards'])
+    _configure_table_helper(
+        connection, table, replicas=replicas, shards=shards)
+
+_configure_table_shards = partial(_configure_table, replicas=None)
+_configure_table_replicas = partial(_configure_table, shards=None)
```

Comment:
@sohkai yes, I agree with your main proposal, as this what I had originally written, and starred at it for several minutes as I was wondering if the duplication of logic could be avoided, because everything is identical except for one line:

```python
replicas = len(table_config['shards'][0]['replicas'])
```

versus

```python
shards = len(table_config['shards'])
```

moreover, these two lines are almost identical except for the value they fetch out of the config `dict`